contracts/HundredVesting.sol

```diff
@@ -23,55 +23,50 @@ contract HundredVesting is Ownable {
+        return amount < 0 ? 0 : amount;
     }
     
-    function vesting (address vester, uint256 amount) external {
-        require(vester != address(0), "Invalid address");
+    function vesting (address vestedAddress, uint256 amount) external {
         require(amount != 0, "Amount should bigger than 0");
 
-        Hundred.transferFrom(msg.sender, address(this), amount);
-        UserInfo memory user = addresses[vester];
+        Hundred.safeTransferFrom(msg.sender, address(this), amount);
+        UserInfo memory user = addresses[vestedAddress];
         if (user.amount != 0) {
             user.amount = user.amount - user.claimedAmount + amount;
             user.timestamp = block.timestamp;
             user.claimedAmount = 0;
         } else {
             user = UserInfo({ amount: amount, timestamp: block.timestamp, claimedAmount: 0 });
         }
-        user.timestamp = block.timestamp;
-        addresses[vester] = user;
+        addresses[vestedAddress] = user;
     }
 
-    function claim (address vester) public {
-        require(vester != address(0), "Invalid address");
-        uint256 amount = getRemainedAmount(vester);
+    function claim () public {
+        require(msg.sender != address(0), "Invalid address");
+        uint256 amount = getRemainedAmount(msg.sender);
         require(amount != 0, "No claimable hundred token");
 
-        addresses[vester].claimedAmount = addresses[vester].claimedAmount + amount;
-        Hundred.transfer(msg.sender, amount);
+        Hundred.safeTransfer(msg.sender, amount);
+        addresses[msg.sender].claimedAmount = addresses[msg.sender].claimedAmount + amount;
     }
 
-    function getClaimableAmount (address vester) public view returns(uint) {
-        require(vester != address(0), "Invalid address");
-        UserInfo memory user = addresses[vester];
+    function getClaimableAmount () public view returns(uint) {
+        UserInfo memory user = addresses[msg.sender];
         require(user.amount != 0, "It's not a vested account");
-        return getRemainedAmount(vester);
+        return getRemainedAmount(msg.sender);
     }
 
-    function getClaimedAmount (address vester) public view returns(uint) {
-        require(vester != address(0), "Invalid address");
-        UserInfo memory user = addresses[vester];
+    function getClaimedAmount () public view returns(uint) {
+        UserInfo memory user = addresses[msg.sender];
         require(user.amount != 0, "It's not a vested account");
         return user.claimedAmount;
     }
 
-    function getVestedAmount (address vester) public view returns(uint) {
-        require(vester != address(0), "Invalid address");
-        UserInfo memory user = addresses[vester];
+    function getVestedAmount () public view returns(uint) {
+        UserInfo memory user = addresses[msg.sender];
         require(user.amount != 0, "It's not a vested account");
```

Comment:
I believe you don't need require here, if not available it will return `0` which is a valid value